### PR TITLE
Add captcha for user registration

### PR DESCRIPTION
**Pull request by AliasQli**
This pr implements #1090 as a gsoc project mentored by @gbaz. Looks like this:

Note:
  - This pr depends on the haskell package `gd`, which in turn requires some system packages to be installed. See `shell.nix` for newly added dependencies.
  - This pr modifies the template file `SignupRequest.html.st`. The file in this repo seems to be different from that on the working hackage server, so when deployed, modify the file correspondingly.

I know little about html and javascript, so please point out if I got something wrong.

### CODE REVIEW

**Comment by AliasQli** [comment]
Ouch, it's the dependencies! I'll fix them soon.

**Comment by AliasQli** [comment]
I've tested it in my own repo's CI.

File: datafiles/templates/UserSignupReset/SignupRequest.html.st [hunk 1]
```diff
@@ -48,4 +56,27 @@ can set your password and activate your account.
 </form>
 
 </div>
+<script type="text/javascript">
+var image = document.getElementById("image");
+var timestamp = document.getElementById("timestamp");
+var hash = document.getElementById("hash");
+
+function changeCaptcha() {
+  var xmlHttp = new XMLHttpRequest();
```
Comment (freizl) [inline datafiles/templates/UserSignupReset/SignupRequest.html.st:65]: Maybe use [fetch](https://caniuse.com/fetch) ?

Comment (AliasQli) [reply datafiles/templates/UserSignupReset/SignupRequest.html.st:65]: Sorry for being busy these days. I'm not familiar with javascript; is there a reason for preferring fetch to XMLHttpRequest?

Comment (freizl) [reply datafiles/templates/UserSignupReset/SignupRequest.html.st:65]: Generally it is [easier to use](https://medium.com/beginners-guide-to-mobile-web-development/the-fetch-api-2c962591f5c) than `XMLHttpRequest`.
It doesn't support IE, but who cares IE. :D

**Pull request merged**